pkg/cloudprovider/providers/openstack/openstack_volumes.go

```diff
@@ -215,40 +215,50 @@ func (os *OpenStack) AttachDisk(instanceID, volumeID string) (string, error) {
 	if err != nil {
 		return "", err
 	}
+
 	if volume.Status != VolumeAvailableStatus {
```

Comment:
+1
If the volume had already been attached to the instance, then it will returned at https://github.com/jamiehannaford/kubernetes/blob/92c59ed84e56ebc14e59ad321f34746134482bf5/pkg/cloudprovider/providers/openstack/openstack_volumes.go#L222 but not supposed to be returned at https://github.com/jamiehannaford/kubernetes/blob/92c59ed84e56ebc14e59ad321f34746134482bf5/pkg/cloudprovider/providers/openstack/openstack_volumes.go#L228 , right?